sklearn/preprocessing/_target_encoder.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
```

Comment:
why a new file instead of using `_encoders.py`?

Reply:
I have a slight preference to have a new file to have files with less lines which makes it slightly easier to maintain.

Reply:
I agree about long files not being optimal, but this introduces an inconsistency for no obvious reason